server/src/main/java/org/opentosca/toscana/plugins/cloudformation/handler/EnvironmentHandler.java

```diff
@@ -0,0 +1,121 @@
+    private static final String ETC = "/etc/";
+    private static final String ETC_ENVIRONMENT = ETC + "environment";
+    private static final String ETC_APACHE2_ENVVARS = ETC + "apache2/envvars";
+    private static final String SET_ENV = "set-env-";
+    private static final String EXPORT = "export ";
+    private static final String AND = " && ";
+
+    public static final String APACHE_ENV_IMPORT = ECHO
+        + ". " + ETC_ENVIRONMENT + " "
+        + REDIRECT_OUTPUT + ETC_APACHE2_ENVVARS + AND + APACHE_RESTART_COMMAND;
+
+    private CloudFormationModule cfnModule;
+    private PluginFileAccess access;
+    private final Logger logger;
+    private final Map<String, Map<String, String>> environmentMap;
+
+    public EnvironmentHandler(CloudFormationModule cfnModule, Logger logger) {
+        this.cfnModule = cfnModule;
+        this.access = cfnModule.getFileAccess();
+        this.environmentMap = cfnModule.getEnvironmentMap();
+        this.logger = logger;
+    }
+
+    /**
+     Writes the necessary scripts to set the environment variables, adds them to the instances and cop
+     */
+    public void handleEnvironment() throws IOException {
+        logger.debug("Handling environment variables.");
+        writeSetEnvScripts();
+        addSetEnvScriptsToInstances();
+        addSetEnvScriptsToFileUploads();
+    }
+
+    /**
+     Writes the setEnv scripts for each Instance.
+     */
+    private void writeSetEnvScripts() throws IOException {
+        logger.debug("Writing setEnv scripts.");
+        for (Map.Entry<String, Map<String, String>> instanceEnvironment : environmentMap.entrySet()) {
+            CloudFormationScript setEnvScript = new CloudFormationScript(access, SET_ENV + instanceEnvironment.getKey());
+            for (Map.Entry<String, String> environmentVariable : instanceEnvironment.getValue().entrySet()) {
+                // Build the command to write environment variable to /etc/environment
+                setEnvScript.append(ECHO + EXPORT
+                    + environmentVariable.getKey() + "=$" + environmentVariable.getKey() + " "
+                    + REDIRECT_OUTPUT + ETC_ENVIRONMENT);
+            }
+        }
+    }
+
+    /**
+     Adds the setEnv scripts to their respective instances and adds commands to execute them.
+     */
+    private void addSetEnvScriptsToInstances() {
+        logger.debug("Adding setEnv scripts to Instances.");
+        for (Map.Entry<String, Map<String, String>> instanceEnvironment : environmentMap.entrySet()) {
+            String nodeName = instanceEnvironment.getKey();
+            String filePath = SET_ENV + instanceEnvironment.getKey() + ".sh";
+            String cfnSource = getFileURL(cfnModule.getBucketName(), filePath);
+
+            CFNFile cfnFile = new CFNFile(ABSOLUTE_FILE_PATH + filePath)
+                .setSource(cfnSource)
+                .setMode(MODE_500) //TODO Check what mode is needed (read? + execute?)
```

Comment:
still TODO?

Reply:
At this point, the modes and owners are probably correct. When we find the time, we'll double check these again until then we'll keep #406 open.